workflow/setup_expt.py

```diff
@@ -229,14 +229,15 @@ def fill_EXPDIR(inputs):
 def update_configs(host, inputs):
 
     def _update_defaults(dict_in: dict) -> dict:
-        defaults = dict_in.pop('defaults', dict())
-        return AttrDict(defaults, **dict_in)
+        defaults = dict_in.pop('defaults', AttrDict())
```

Comment:
Are we sure defaults shouldn't just be always be cast as an `AttrDict`. Would avoid running into the same error again the next time we send `_update_defaults` a non-AttrDict dict.

Reply:
`_update_defaults` is a private method and it is unlikely we will use it generally elsewhere. 
It had a very specific use case here.

Reply:
We do need AttrDict to be the overloading Dict for all YAML configs to setup though I guess for everything else to work?  Anyway, it worked and I've coded against it and my PR is now pending on this.

Reply:
I think the best way would be for `YAMLFile` to return an `AttrDict` object, which it should.
I think we just need to `YAMLFile.parse_yaml` to return `AttrDict` instead of `dict`
I think that is backward compatible (but I would hold that for a PR of its own).